src/main/java/com/mathworks/ci/MatlabBuilder.java

```diff
@@ -148,8 +150,9 @@ public DescriptorExtensionList<TestRunTypeList, Descriptor<TestRunTypeList>> get
          * descriptor class.
          */
 
-
+        @POST
         public FormValidation doCheckMatlabRoot(@QueryParameter String matlabRoot) {
+            Jenkins.get().checkPermission(Permission.CONFIGURE);
```

Comment:
This will be incorrect for some configurations. Please use `@AncestorInPath` here to obtain the context, and then check for `Item.CONFIGURE` permission on that context. See https://www.jenkins.io/doc/developer/security/form-validation/#checking-permissions for example.

Reply:
Done.